src/js/components/common/minard-link.tsx

```diff
@@ -3,46 +3,56 @@ import { Link } from 'react-router';
 
 import { Branch } from '../../modules/branches';
 import { Comment } from '../../modules/comments';
-import { Commit } from '../../modules/commits';
 import { Deployment } from '../../modules/deployments';
 import { Project } from '../../modules/projects';
 
 interface Props {
-  deployment?: Deployment;
-  preview?: Deployment;
-  commit?: Commit;
-  branch?: Branch | string;
-  project?: Project | string;
-  comment?: Comment;
-  openInNewWindow?: boolean;
+  rawDeployment?: {
+    deployment: Deployment;
```

Comment:
That has two benefits:
1. I don't need to remember which types of entities I have to pass the entity in directly and which types require a "wrapper" object.
2. Easily add options in the future.